com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoopDef.hlsl

```diff
@@ -343,16 +346,17 @@ EnvLightData FetchEnvLight(uint index)
     return _EnvLightDatas[index];
 }
 
-// In the first 8 bits of the target we store the max fade of the contact shadows as a byte
+// In the first bits of the target we store the max fade of the contact shadows as a byte.
+//By default its 8 bits for the fade and 24 for the mask, please check the LightLoop.cs definitions.
 void UnpackContactShadowData(uint contactShadowData, out float fade, out uint mask)
 {
-    fade = float(contactShadowData >> 24) / 255.0;
-    mask = contactShadowData & 0xFFFFFF; // store only the first 24 bits which represent
+    fade = float(contactShadowData >> CONTACT_SHADOW_MASK_BITS) / ((float)CONTACT_SHADOW_FADE_MASK);
```

Comment:
So yeah contact shadow are gonna be awkward :D I wonder if we should change a bit how we do it later on. 

Now if we have only one light that has contact shadow, but that happens to have an index bigger than 24 in the list it will not show up. 

Even worse if that light changes depending on culling from say 20 and 25 as index, it will "flicker" contact shadows :/

Reply:
I don't remind exactly but aren't we sorting light by shadow casting? i.e the first set of light in the list currently cast shadow? increasing the likelihood that we have shadow on the first 24 light index. Guess something to test

Reply:
So there is no sorting by shadow casting as far as I can tell. We sort by light type on the entire frame, then we sort per light index per tile.
The way it works is that we allocate all contact shadows from scratch every frame on all lights on the view. Not on a per tile basis.

The very first 24 lights that *Have* shadows enabled get it, and the allocation is on the CPU.  Note that we can have 1 light with contact shadows followed by 32 that dont, and then 24 more that have contact shadows. In this case it will be perfectly fine. 

The only way shadow allocation can vary is if a new light enters the view (passes culling).

For example, we have in the frame 512 lights in view. The first one and last 23 ones have contact shadows enabled.
 * In this situation we will never get flickering
 
 Now lets assume we have 1 more light with contact shadow, now we have a total of 25.
 If all lights are on view, it wont flicker either because the sorting in the CPU remains deterministic.
 
 Hope this makes sense.

Reply:
Yeah stuff changed and forgot, used to be that each bit of the 24 bit mapped to the index of the light in the tile (and there 24 was convenient for that :P )